### ISSUE TITLE
Pass a context to every kopf call/handler

### ISSUE THREAD

**Issue by krish7919** [zalando-incubator/kopf#347]
## Question

I would like to embed kopf in an application to perform some extra cluster functions as a response to events.

I can always use a global variable.
Actually, I _have to_ go the embedding way as I want to build wrappers to take some custom arguments as input, validate them and then start the kopf asyncio loop. Plus I have to pass these params around as every handler would need them.

This should be a pretty common ask, and I am not sure how is that done currently.

## Checklist

- [x] I have read the [documentation](https://kopf.readthedocs.io/en/latest/) and searched there for the problem

- [x] I have searched in the [GitHub Issues](https://github.com/zalando-incubator/kopf/issues?utf8=%E2%9C%93&q=) for similar questions

## Keywords

"kopf pass context to handlers" on Google and in docs.

**Comment by nolar** [comment]
@krish7919 Just to clarify: do you mean an arbitrary object of arbitrary class, which is _passed through_ the call stack from the owning application down to the handlers (all of them), right? Something like a `memo` kwarg (per-resource container for arbitrary keys/fields), but global and of external origin?

**Comment by krish7919** [comment]
Precisely. In Go, this is the `context`, or a `struct` with the values required.

**Comment by mnaser** [comment]
This is something that's interesting for our usage too, especially when we're constantly trying to connect to the Kuberentes API..

**Comment by eshepelyuk** [comment]
This issue also may be related to my question in #367
